source/Files.cpp

```diff
@@ -101,42 +88,38 @@ void Files::Init(const char * const *argv)
 	{
 		// Find the path to the resource directory. This will depend on the
 		// operating system, and can be overridden by a command line argument.
-		char *str = SDL_GetBasePath();
-		if(!str)
+		resources = filesystem::current_path();
+		if(resources.empty())
 			throw runtime_error("Unable to get path to resource directory!");
 
-		resources = str;
-		SDL_free(str);
+		if(exists(resources))
+			resources = filesystem::canonical(resources);
+
+		#if defined __linux__ || defined __FreeBSD__ || defined __DragonFly__
+			// Special case, for Linux: the resource files are not in the same place as
+			// the executable, but are under the same prefix (/usr or /usr/local).
+			static const filesystem::path LOCAL_PATH = "/usr/local/";
+			static const filesystem::path STANDARD_PATH = "/usr/";
+			static const filesystem::path RESOURCE_PATH = "share/games/endless-sky/";
+			if(resources.string().starts_with(LOCAL_PATH.string()))
+				resources = LOCAL_PATH / RESOURCE_PATH;
+			else if(resources.string().starts_with(STANDARD_PATH.string()))
+				resources = STANDARD_PATH / RESOURCE_PATH;
+		#endif
```

Comment:
```suggestion
#if defined __linux__ || defined __FreeBSD__ || defined __DragonFly__
		// Special case, for Linux: the resource files are not in the same place as
		// the executable, but are under the same prefix (/usr or /usr/local).
		static const string LOCAL_PATH = "/usr/local/";
		static const string STANDARD_PATH = "/usr/";
		static const filesystem::path RESOURCE_PATH = "share/games/endless-sky/";
		if(resources.string().starts_with(LOCAL_PATH))
			resources = LOCAL_PATH / RESOURCE_PATH;
		else if(resources.string().starts_with(STANDARD_PATH))
			resources = STANDARD_PATH / RESOURCE_PATH;
#endif
```

This avoids unnecessary conversion / uses the right identation